src/main/java/com/desiato/puresynth/configurations/AuthTokenFilter.java

```diff
@@ -0,0 +1,65 @@
+package com.desiato.puresynth.configurations;
+
+import com.desiato.puresynth.services.AuthenticationService;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+import java.util.Collections;
+
+@Component
+public class AuthTokenFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private AuthenticationService authService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain)
+            throws ServletException, IOException {
+        String token = request.getHeader("Auth-Token");
+
+        if (request.getRequestURI().equals("/api/login") || request.getRequestURI().equals("/api/user")) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        if (token == null) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            response.getWriter().write("Unauthorized");
+            return;
+        }
+
+        try {
+            if (!authService.isUserAuthenticated(token)) {
+                response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+                response.getWriter().write("Unauthorized: Invalid token");
+                return;
+            }
+
+            UserDetails userDetails = authService.loadUserByToken(token);
```

Comment:
You have 3 checks, which duplicate a lot of code when the user is not authenticated. You should encapsulate this inside `AuthenticationService`. There is no reason you should make two call one-by-one to it:
```
authService.isUserAuthenticated(token)
authService.loadUserByToken(token)
```

Instead you should have a call like:
```
Optional<UserDTO> userDTO = authService.authenticateByToken(token)
```
Inside it the service should compute what's necessary.